--- afl-0.50b.old/Makefile	2014-11-16 16:16:11.000000000 +0000
+++ afl-0.50b/Makefile	2014-11-16 16:19:13.328069379 +0000
@@ -21,7 +21,7 @@
 
 PROGS       = afl-gcc afl-as afl-fuzz afl-showmap
 
-CFLAGS     += -O3 -Wall -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign \
+override CFLAGS += -O3 -Wall -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign \
 	      -DAFL_PATH=\"$(HELPER_PATH)\" -DVERSION=\"$(VERSION)\"
 
 ifneq "$(HOSTNAME)" "raccoon"
