diff -ur afl-2.00b/Makefile afl-2.00b-cflags/Makefile
--- afl-2.00b/Makefile	2016-02-12 06:06:14.000000000 +0000
+++ afl-2.00b-cflags/Makefile	2016-02-15 12:04:56.690784941 +0000
@@ -26,7 +26,7 @@
 SH_PROGS    = afl-plot afl-cmin afl-whatsup
 
 CFLAGS     ?= -O3 -funroll-loops
-CFLAGS     += -Wall -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign \
+override CFLAGS += -Wall -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign \
 	      -DAFL_PATH=\"$(HELPER_PATH)\" -DDOC_PATH=\"$(DOC_PATH)\" \
 	      -DBIN_PATH=\"$(BIN_PATH)\" -DVERSION=\"$(VERSION)\"
 
Only in afl-2.00b-cflags: Makefile.orig
