diff -ur afl-1.71b.old/Makefile afl-1.71b/Makefile
--- afl-1.71b.old/Makefile	2015-04-20 02:17:54.000000000 +0100
+++ afl-1.71b/Makefile	2015-04-23 21:34:41.888772510 +0100
@@ -25,7 +25,7 @@
 PROGS       = afl-gcc afl-as afl-fuzz afl-showmap afl-tmin afl-gotcpu
 
 CFLAGS     ?= -O3 -funroll-loops
-CFLAGS     += -Wall -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign \
+override CFLAGS += -Wall -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign \
 	      -DAFL_PATH=\"$(HELPER_PATH)\" -DDOC_PATH=\"$(DOC_PATH)\" \
 	      -DBIN_PATH=\"$(BIN_PATH)\" -DVERSION=\"$(VERSION)\"
 
Only in afl-1.71b: Makefile.orig
